### PR TITLE
Feature/rolling spectrums

### PR DESCRIPTION
**Pull request by shanlyMIDE**
Lots in this!

- Added `rolling_fft` to `endaq.calc.fft`
- Added `rolling_psd` to `endaq.calc.psd`
- Added `rolling_shock_spectrum` to `endaq.calc.shock`
- Allowed endaq.calc.shock.shock_spectrum to calculate the freqs for you (instead of requiring other function call)
- Added `spectrum_over_time` to `endaq.plot.plots`
- Fixed `octave_spectrogram`

This would close / address: #189, #35, #156, 

[This specific branch is tested in this colab with lots of examples](https://colab.research.google.com/drive/11YWFw4Y5ymqr-T9nfgxsXSDIa9whBbLS?usp=sharing)

### CODE REVIEW

File: endaq/calc/shock.py [hunk 1]
```diff
@@ -405,21 +405,25 @@ def relative_displacement_static(accel: pd.DataFrame, omega: float, damp: float
...
+    init_freq: float = 0.5,
+    bins_per_octave: float = 12.0,
     damp: float = 0.0,
     mode: typing.Literal["srs", "pvss"] = "srs",
     two_sided: bool = False,
     aggregate_axes: bool = False,
 ) -> pd.DataFrame:
     """
     Calculate the shock spectrum of an acceleration signal.
-
```
Comment (CFlaniganMide) [inline endaq/calc/shock.py:416]: This newline is critical to formatting the docs.

File: endaq/plot/plots.py [hunk 2]
```diff
@@ -4,14 +4,16 @@
 import pandas as pd
 import plotly.graph_objects as go
 import plotly.express as px
+from plotly import colors
 import scipy.spatial.transform
 from plotly.subplots import make_subplots
 from scipy import signal, spatial
 from typing import Optional
 from collections.abc import Container
+import datetime
```
Comment (CFlaniganMide) [inline endaq/plot/plots.py:13]: Builtin modules should be imported before 3rd party modules.

**Review by CFlaniganMide — CHANGES_REQUESTED** [review]
Not going to point out all the same stuff as the other PR (yet), just calling out a few things that are unique to this one.

**Comment by shanlyMIDE** [comment]
@CFlaniganMide let me know when you want to review this, I have the unit tests in there now and am curious how well written you think they are?

File: tests/calc/test_fft.py [hunk 3]
```diff
@@ -62,48 +62,47 @@ def test_fft(self, normalization, output, n):
...
 
         fxx = np.fft.rfft(data)
 
         n = len(data)
 
-        scales = {"unit": 2./n, "forward": 1./n, "backward": 1., "ortho": np.sqrt(1./n)}
+        scales = {"unit": 2. / n, "forward": 1. / n, "backward": 1., "ortho": np.sqrt(1. / n)}
 
-        scale = scales.get(normalization, 2./n)
+        scale = scales.get(normalization, 2. / n)
```
Comment (CFlaniganMide) [inline tests/calc/test_fft.py:105]: also this

File: endaq/calc/psd.py [hunk 4]
```diff
@@ -335,3 +336,263 @@ def vc_curves(
...
+
+    # Windowing, Default to Boxcar
+    if "window" not in kwargs:
+        kwargs["window"] = "boxcar"
+
+    # Determine Scipy.Spectrogram Stats
+    fs = 1 / utils.sample_spacing(df)
+    nperseg = int(fs / bin_width)
+    num_f = int(len(df) / fs / bin_width)
+    noverlap = nperseg - (nperseg * num_f / num_slices)
```
Comment (CFlaniganMide) [inline endaq/calc/psd.py:532]: use integer division here to ensure data types
```suggestion
    noverlap = nperseg - (nperseg*num_f//num_slices)
```

**Comment by shanlyMIDE** [comment]
@CFlaniganMide I think I've done everything now correctly, but let me know of course if not!

**Pull request merged**